sdk/ai/azure-ai-inference/README.md

```diff
@@ -0,0 +1,476 @@
+
+Use the model inference client library to:
+
+* Authenticate against the service
+* Get information about the model
+* Do chat completions
+* Get text embeddings
+<!-- * Get image embeddings -->
+
+Note that for inference using OpenAI models hosted on Azure, you should be using the official [OpenAI Python client library](https://github.com/openai/openai-python) in product code instead of this client. However, for development and evaluation purposes (comparing OpenAI models to other models in the Azure AI Studio catalog), you can use the azure-ai-inference Python client library with Azure OpenAI endpoints, as shown [in this sample](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-inference/samples/sample_chat_completions_azure_openai.py).
+
+[Product documentation](https://learn.microsoft.com/azure/ai-studio/reference/reference-model-inference-api)
+| [Samples](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/ai/azure-ai-inference/samples)
+| [API reference documentation](https://aka.ms/azsdk/azure-ai-inference/python/reference)
+| [Package (Pypi)](https://aka.ms/azsdk/azure-ai-inference/python/package)
+| [SDK source code](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/ai/azure-ai-inference/azure/ai/inference)
+
+## Getting started
+
+### Prerequisites
+
+* [Python 3.8](https://www.python.org/) or later installed, including [pip](https://pip.pypa.io/en/stable/).
+* An [Azure subscription](https://azure.microsoft.com/free).
+* An [AI Model from the catalog](https://ai.azure.com/explore/models) deployed through Azure AI Studio.
+* To construct the client library, you will need to pass in the endpoint URL. The endpoint URL has the form `https://your-deployment-name.your-azure-region.inference.ai.azure.com`, where `your-deployment-name` is your unique model deployment name and `your-azure-region` is the Azure region where the model is deployed (e.g. `eastus2`).
+* Depending on your model deployment and authentication preference, you either need a key to authenticate against the service, or Entra ID credentials. The key is a 32-character string.
+
+### Install the package
+
+To install the Azure AI Inferencing package use the following command:
+
+```bash
+pip install azure-ai-inference
+```
+
+To update an existing installation of the package, use:
+
+```bash
+pip install --upgrade azure-ai-inference
+```
+
+## Key concepts
+
+### Create and authenticate a client directly, using key
+
+The package includes two clients `ChatCompletionsClient` and `EmbeddingsClient`<!-- and `ImageGenerationClients`-->. Both can be created in the similar manner. For example, assuming `endpoint` and `key` are strings holding your endpoint URL and key, this Python code will create and authenticate a synchronous `ChatCompletionsClient`:
+
+```python
```

Comment:
I would suggest starting with a zero-to-hero scenario off the bat. Possibly using

```python
creds = azure.identity.DefaultAzureCredential(exclude_interactive_browser_credential=False)
```

which will pop a browser window if the user isn't already logged in... The goal would be to have something that I can copy/paste and run and not have a bunch of things in between the code blocks. If you need to comment on something, commenting in the code can be done instead of splitting code blocks with markdown blocks.

Reply:
I updated all examples of using DefaultAzureCredential() to DefaultAzureCredential(exclude_interactive_browser_credential=False).

Reply:
Regarding _"The goal would be to have something that I can copy/paste and run and not have a bunch of things in between the code blocks"_ -- I agree! and if you look at the chat completion and text embeddings example sections, you will see fully runnable code there (creating the client and doing chat completions/embeddings). You still need to provide the endpoint and key though.

Reply:
I was thinking about perhaps showing reading endpoint url and key directly from environment variables, in the constructor of the clients. Code in README.me is picked up from the samples. In the samples I do read url and key from environment variables, but it is in a separate "try" block at the start, so I can have a better error message when one of the environment variables is missing. But I don't want that in the code snippets in README.md. Bottom line: I need to think about this some more. For now, as there is urgency in getting a first release out, I prefer to keep the code snippets in  README.md as is. But may change that in the next release based on additional discussion with you and feedback. I hope that's okay with you. Thanks!